packages/anoncreds-rs/tests/InMemoryTailsFileService.ts

```diff
@@ -0,0 +1,60 @@
+import type { AnonCredsRevocationRegistryDefinition } from '@aries-framework/anoncreds'
+import type { AgentContext } from '@aries-framework/core'
+
+import { BasicTailsFileService } from '@aries-framework/anoncreds'
+
+export class InMemoryTailsFileService extends BasicTailsFileService {
+  private tailsFilePaths: Record<string, string> = {}
+
+  public async uploadTailsFile(
```

Comment:
This `tailsLocation` thing is a bit confusing, as the output from the library is a RevocationRegistryDefinition whose tailsLocation is actually the local file path, and then we should update it with the actual URL path before storing in the wallet and publishing to the VDR.

Besides this explanation, I think you are right and it will be better to not use the revocation registry definition object but its ID and either the local tails file path or the file contents itself. At first sight I'd prefer to pass the path, as usually libraries like fetch or axios do work well with file streams and that could be more efficient if handled by lower level modules.